RecoEgamma/EgammaTools/src/PhotonIDHelper.cc

```diff
@@ -0,0 +1,61 @@
+#include "RecoEgamma/EgammaTools/interface/PhotonIDHelper.h"
+
+#include <iostream>
+
+PhotonIDHelper::PhotonIDHelper(const edm::ParameterSet  & iConfig,edm::ConsumesCollector && iC):
+        eeRecHitInputTag_(iConfig.getParameter<edm::InputTag> ("EERecHits") ),
+        fhRecHitInputTag_(iConfig.getParameter<edm::InputTag> ("FHRecHits") ),
+        bhRecHitInputTag_(iConfig.getParameter<edm::InputTag> ("BHRecHits") ),
+        dEdXWeights_(iConfig.getParameter<std::vector<double> >("dEdXWeights"))
+{
+    isoHelper_.setDeltaR(iConfig.getParameter<double>("photonIsoDeltaR"));
+    isoHelper_.setNRings(iConfig.getParameter<unsigned int>("photonIsoNRings"));
+    isoHelper_.setMinDeltaR(iConfig.getParameter<double>("photonIsoDeltaRmin"));
+
+    recHitsEE_ = iC.consumes<HGCRecHitCollection>(eeRecHitInputTag_);
+    recHitsFH_ = iC.consumes<HGCRecHitCollection>(fhRecHitInputTag_);
+    recHitsBH_ = iC.consumes<HGCRecHitCollection>(bhRecHitInputTag_);
```

Comment:
same comment as for the electron: to add HGC to the name because this class can not be used for all photons